plugins/VstBase/RemoteVstPlugin.cpp

```diff
@@ -1801,86 +1773,71 @@ intptr_t RemoteVstPlugin::hostCallback( AEffect * _effect, int32_t _opcode,
 			return 0;
 
 		case audioMasterGetTime:
+		{
 			SHOW_CALLBACK( "amc: audioMasterGetTime\n" );
 			// returns const VstTimeInfo* (or 0 if not supported)
 			// <value> should contain a mask indicating which
 			// fields are required (see valid masks above), as some
 			// items may require extensive conversions
 
-			// Shared memory was initialised? - see song.cpp
-			//assert( __plugin->m_vstSyncData != nullptr );
+			const auto syncData = __plugin->getVstSyncData();
+			assert(syncData != nullptr);
```

Comment:
You're making RemoteVstPlugin crash when VST sync data is available. It's acceptable, but could you give some explanations?

Reply:
See my other comment for the reasons behind this. The assertion here is because it should not be possible for `syncData` to be null here - the `IdSyncKey` message should have been sent by LMMS, and if the shared memory failed to attach, the remote plugin should have terminated at that point. I generally try to use exceptions to handle runtime errors (such as failure to attach shared memory), and assertions to handle logic errors (such as here, where code elsewhere should ensure this condition is true).